test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -557,24 +557,26 @@ def skip_smddataparallel_test(
 
 
 @pytest.fixture(autouse=True)
-def skip_p5_tests(instance_type, processor, ecr_image):
-    if "p5." in instance_type:
+def skip_p5_tests(instance_type, efa_instance_type, processor, ecr_image):
+    if "p5." in instance_type or "p5." in efa_instance_type:
```

Comment:
If you're checking both `instance_type` and `efa_instance_type`, I'd recommend checking the parameter only when it is actually defined in the test.

You can check if the fixture is actually parametrized through `if fixture in request.fixturenames`, and then get its value through `request.getfixturevalue("{fixture}")`.

I think there is potential for this `skip_p5_tests` fixture to behave in unexpected ways.

Have you tested this fixture function out with all possible combinations of relevant instance types for `instance_type` and `efa_instance_type`?

Reply:
Fix the tests and use request to check parameters instead.

I'm not sure what you mean by testing out the possible combinations of relevant instance types though.